### DISCUSSION TITLE
[Feature] 近期QQ中的双击头像"戳了戳"、“拍了拍”动作

### DISCUSSION THREAD

**Discussion by yyuueexxiinngg** [howmanybots/onebot#85] [想法 / 建议]
近期QQ群组中也支持了双击用户头像进行“戳一戳”的动作 （类似于微信中的“拍一拍”）

此动作已在[go-cqhttp](https://github.com/Mrs4s/go-cqhttp)中实现为[[CQ:poke]](https://github.com/Mrs4s/go-cqhttp/blob/master/docs/cqhttp.md#%E6%88%B3%E4%B8%80%E6%88%B3), 但此实现与OneBot标准中的[[CQ:poke]](https://github.com/howmanybots/onebot/blob/master/v11/specs/message/segment.md#%E6%88%B3%E4%B8%80%E6%88%B3)冲突

此"戳一戳"不太符合"消息"的定义, 而更像"事件":
- 触发行为并非在消息框或消息框下的消息动作 *(如私聊时点加号可发送的旧版"戳一戳")*
- 接收此行为时的显示与事件类似 *(显示在中央事件位置而非左侧消息位置)*

OneBot中已将此动作定性为事件, 见[群内戳一戳](https://github.com/howmanybots/onebot/blob/master/v11/specs/event/notice.md#%E7%BE%A4%E5%86%85%E6%88%B3%E4%B8%80%E6%88%B3), 或许此行为的发送可设计为API而非CQ码 *(如设计为CQ码可能发生夹带在消息链中的情况)*

欢迎大家参与讨论

**Comment by stdrc** [comment]
我倾向于使用消息段发送，毕竟它其实也不是一种典型的 API（目前的 API 似乎主要都是需要在群管理、好友资料页面点的操作），它还是在聊天记录页面触发的

接收暂时没什么更好的想法

**Comment by yyuueexxiinngg** [comment]
> 我倾向于使用消息段发送

看了下[`mirai`的实现](https://github.com/mamoe/mirai/blob/dev/mirai-core-api/src/commonMain/kotlin/message/action/Nudge.kt), 也是将其当作了消息, 目前`onebot-kotiln`已支持作为消息段发送, 不过不确定其类型命名, 暂时参考`mirai`将其命名为了`nudge`

**Comment by Ero-Cat** [comment]
poke感觉跟准确一些........ 如果拿不准...可以消息与事件都做一个接口   ，不过我更倾向于消息

**Comment by yyuueexxiinngg** [comment]
> poke感觉跟准确一些........ 如果拿不准...可以消息与事件都做一个接口 ，不过我更倾向于消息

poke的话与原有的表情戳一戳冲突了...

**Comment by Ero-Cat** [comment]
> > poke感觉跟准确一些........ 如果拿不准...可以消息与事件都做一个接口 ，不过我更倾向于消息
> 
> poke的话与原有的表情戳一戳冲突了...

那这样就有问题了....以后说不定还有更多关于戳一戳的功能，不能总换个单词吧,,,,可以[CQ:poke_head]?

**Comment by stdrc** [comment]
不如合并成一种（都是一种“唤起注意”操作），然后在消息段用 `style` 参数区分，我甚至觉得 shake（现在还有 shake 吗，电脑上点窗口抖动的那种）也应该和这些作为同类对待

**Comment by ExerciseBook** [comment]
（其实我感觉他把 shake 给套了个壳然后出来骗钱）

**Comment by takayama-lily** [comment]
目前的shake和poke虽说是私聊里的，但好像可以用bot在群里发

**Comment by takayama-lily** [comment]
[CQ:poke,type=126,id=2003] 现在的戳一戳
[CQ:poke,qq=123456] 双击头像戳一戳
两者都是poke但参数不同，实际并无冲突
